src/backend/distributed/relay/relay_event_utility.c

```diff
@@ -701,3 +702,60 @@ shard_name(PG_FUNCTION_ARGS)
 	AppendShardIdToName(&relationName, shardId);
 	PG_RETURN_TEXT_P(cstring_to_text(relationName));
 }
+
+
+/*
+ * qualified_shard_name is a UDF that returns the name of a shard as a
+ * quoted, schema-qualified identifier.
+ */
+Datum
+qualified_shard_name(PG_FUNCTION_ARGS)
+{
+	Oid relationId = InvalidOid;
+	int64 shardId = 0;
+	char *relationName = NULL;
+	Oid schemaId = InvalidOid;
+	char *schemaName = NULL;
+	char *qualifiedName = NULL;
+
+	if (PG_ARGISNULL(0))
+	{
+		ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						errmsg("object_name cannot be null")));
+	}
+	if (PG_ARGISNULL(1))
+	{
+		ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						errmsg("shard_id cannot be null")));
+	}
+
+	relationId = PG_GETARG_OID(0);
+	shardId = PG_GETARG_INT64(1);
```

Comment:
Mostly because shard_name wasn't (this was an external contribution). Should I just make both strict?